src/timeline/widgets/ImageItem.cc

```diff
@@ -83,7 +84,16 @@ ImageItem::ImageItem(QSharedPointer<MatrixClient> client,
         url_                 = QString("%1/_matrix/media/r0/download/%2")
                  .arg(client_.data()->getHomeServer().toString(), media_params);
 
-        setImage(QPixmap(filename));
+        auto null  = data.isNull();
+        auto reset = data->reset();
+        QPixmap p;
+
+        if (!null && reset) {
+                p.loadFromData(data->readAll());
+        } else {
+                p.load(filename);
```

Comment:
Because `filename` no longer identifies a path on disk, this fallback doesn't make sense. An error message (see [`QIODevice::errorString`](https://doc.qt.io/qt-5/qiodevice.html#errorString) should be displayed instead.